Xamarin.PropertyEditing.Mac/Controls/CombinablePropertyEditor.cs

```diff
@@ -0,0 +1,137 @@
+	{
+		public CombinablePropertyEditor ()
+		{
+			base.TranslatesAutoresizingMaskIntoConstraints = false;
+
+			UpdateTheme ();
+		}
+
+		public override NSView FirstKeyView => firstKeyView;
+		public override NSView LastKeyView => lastKeyView;
+
+		internal new CombinablePropertyViewModel<T> ViewModel
+		{
+			get { return (CombinablePropertyViewModel<T>)base.ViewModel; }
+			set { base.ViewModel = value; }
+		}
+
+		List<NSButton> combinableList = new List<NSButton> ();
+		NSView firstKeyView;
+		NSView lastKeyView;
+
+		protected override void HandleErrorsChanged (object sender, DataErrorsChangedEventArgs e)
+		{
+			UpdateErrorsDisplayed (ViewModel.GetErrors (e.PropertyName));
+		}
+
+		protected override void SetEnabled ()
+		{
+			foreach (var item in combinableList) {
+				item.Enabled = ViewModel.Property.CanWrite;
+			}
+		}
+
+		protected override void UpdateErrorsDisplayed (IEnumerable errors)
+		{
+			if (ViewModel.HasErrors) {
+				SetErrors (errors);
+			} else {
+				SetErrors (null);
+				SetEnabled ();
+			}
+		}
+
+		protected override void OnViewModelChanged (PropertyViewModel oldModel)
+		{
+			combinableList.Clear ();
+
+			const float controlHeight = 22;
+
+			// Set our new RowHeight
+			RowHeight = (ViewModel.Choices.Count * controlHeight);
+
+			float top = controlHeight;
+			foreach (var item in ViewModel.Choices) {
+				var BooleanEditor = new NSButton (new CGRect (4, RowHeight - top, Frame.Width - 4, controlHeight)) {
+					AllowsMixedState = true,
+					ControlSize = NSControlSize.Small,
+					Font = NSFont.FromFontName (DefaultFontName, DefaultFontSize),
+					Title = item.Name,
+					TranslatesAutoresizingMaskIntoConstraints = false,
+				};
+				BooleanEditor.SetButtonType (NSButtonType.Switch);
+				BooleanEditor.Activated += SelectionChanged;
+
+				AddSubview (BooleanEditor);
+				combinableList.Add (BooleanEditor);
+				top += controlHeight;
+			}
+
+			// Set our tabable order
+			firstKeyView = combinableList.First ();
+			lastKeyView = combinableList.Last ();
+
+			base.OnViewModelChanged (oldModel);
+		}
+
+		protected override void UpdateValue ()
+		{
+			foreach (var item in combinableList) {
+				if (ViewModel.Choices.Count > 0) {
+					var flag = ViewModel.Choices.FirstOrDefault (x => x.Name == item.Title);
+					if (flag.IsFlagged.HasValue) {
+						item.State = flag.IsFlagged.Value ? NSCellStateValue.On : NSCellStateValue.Off;
+						item.AllowsMixedState = false;
+					} else {
+						item.State = NSCellStateValue.Mixed;
+					}
+				} else {
+					item.State = NSCellStateValue.Off;
+				}
+			}
+		}
+
+		protected override void UpdateAccessibilityValues ()
+		{
+			foreach (var item in combinableList) {
+				item.AccessibilityEnabled = item.Enabled;
+				item.AccessibilityTitle = string.Format (LocalizationResources.AccessibilityCombobox, ViewModel.Property.Name);
+			}
+		}
+
+		void SelectionChanged (object sender, EventArgs e)
+		{
+			var button = sender as NSButton;
+
+			if (button != null) {
```

Comment:
nit: `if (sender is NSButton button) {`

Reply:
hehe I didn't review it after pulling out the code from the other PR. I'll add that in shortly.